contracts/MerkleLib.sol

```diff
@@ -55,7 +55,13 @@ library MerkleLib {
             }
             chunkIdx = chunkIdx / 2;
         }
-        return hash == root;
+        /* NOTICE: Due to our design of PhyAddr, only front 24 bytes
+         *         are valid. We only validate that part.
+         * With no doubt, it introduces some vulnerability compared 
+         * with a standard Merkle validation. It is a trade-off,
+         * if we want to put all meta data into a single bytes32(opcode length)
+         */
+        return bytes24(hash) == bytes24(root);
```

Comment:
Could we add a method like `calculateRootWithProof`, which returns the root so that the comparison logic can be done by application contract?  MerkleLib should not be designed to understand the size of hash.